zp-relayer/workers/sentTxWorker.ts

```diff
@@ -111,16 +108,23 @@ async function handleReverted(
     if (!wj?.id) continue
     waitingJobIds.push(wj.id)
 
-    const { txPayload, traceId } = wj.data
-    const transactions = [txPayload]
+    const { txPayload } = wj.data
+    let reschedulePromise: Promise<any>
 
-    // To not mess up traceId we add each transaction separately
-    const reschedulePromise = poolTxQueue.add(txHash, { transactions, traceId }).then(j => {
-      const newPoolJobId = j.id as string
-      newPoolJobIdMapping[wj.data.poolJobId] = newPoolJobId
-      return newPoolJobId
+    reschedulePromise = poolTxQueue.add(txHash, {
```

Comment:
Maybe it's better if we won't reschedule direct deposit batch, since we already add all deposits back into the queue?

Reply:
There are two cases for DD batch revert:
1. Batch reverted on its own. In this case we probably have some problem with batch itself, and need to re-queue individual deposits back to watcher for re-validation.
2. Batch reverted because of some past tx, which led to tree proof inconsistency. In this case there were no problem with the batch and we can re-queue it directly to `poolTxQueue`. This allows us to use previously computed tx proof. I assumed that tx proof for batches would take a lot of time (was discussed somewhere earlier) and this optimisation will save us some CPU time.

Reply:
But we always do both, don't we? We first request individual deposits back into the reprocessing queue, then requeue whole job into `poolTxQueue`. Will this lead to double processing?

Reply:
>But we always do both, don't we?

No. Here, we check if reverted tx was a DD. If it was, then we add individual deposits for re-validation.
https://github.com/zkBob/zeropool-relayer/blob/0d05c489672536003b3006720c542b370fd65333/zp-relayer/workers/sentTxWorker.ts#L267-L271

And this call re-queues only optimistic txs.
https://github.com/zkBob/zeropool-relayer/blob/0d05c489672536003b3006720c542b370fd65333/zp-relayer/workers/sentTxWorker.ts#L272

Reply:
So, the logic flow for DD batch which reverted because of optimistic state corruption and for DD batch which reverted because of its own reasons is different.